### ISSUE TITLE
HistGradientBoosting 을 활용한 예측

### ISSUE THREAD

**Issue by YounJJ** [Red-Blues-Mid-Projects-OAA/Red-Blue-Mid-Project_OAA#25]
목표치:
  Test Accuracy: 52% 이상
  IC: 0.05 이상
  Train-Test Gap: 25% 미만

고려해야할 사항:
  Optuna 튜닝 대상 파라미터:

      max_depth: 2~6
      learning_rate: 0.01~0.1
      n_estimators: 100~500
      gamma: 0~0.5
      reg_alpha: 0~1.0
      reg_lambda: 0.5~3.0
      subsample: 0.5~0.9

등을 고려해서 HyperParameters Scailing

----

[앙상블 기반의 스트라이드 학습]. 데이터를 버리는 게 아니라 나누어 담는 방식.

5일 간격(1주일) 샘플링 + 5개 모델 앙상블

다음과 같이 5개의 '서로 다른 기억을 가진' 모델

모델 1: 0, 5, 10, 15... 번째 데이터로 학습 (월요일 데이터 위주)
모델 2: 1, 6, 11, 16... 번째 데이터로 학습 (화요일 데이터 위주)
모델 3: 2, 7, 12, 17... 번째 데이터로 학습
모델 4: 3, 8, 13, 18... 번째 데이터로 학습
모델 5: 4, 9, 14, 19... 번째 데이터로 학습

이렇게 하면 각 모델은 타겟이 거의 겹치지 않는 독립적인 시퀀스를 학습하게 되어 Train Acc가 자연스럽게 낮아지고(과적합 해소), 이 5개 모델의 예측치를 평균 내면 전체 데이터의 정보를 100% 활용하면서도 매우 견고한 확률값을 얻을 수 있다고 함.

주의할 점은 refit 단계에서도 같은 앙상블 기반 스트라이드 학습을 적용 해야한다는 것. 그렇지 않으면 재차 학습할때 전체 train, validation set을 다시 학습해버려서 의미가 없음.

**Comment by YounJJ** [comment]
XGBoost와의 Boosting 중복성 및 튜닝 난도로 인한 앙상블 제외.
